keeper-gradle-plugin/src/main/java/com/slack/keeper/KeeperPlugin.kt

```diff
@@ -100,95 +102,184 @@ class KeeperPlugin : Plugin<Project> {
 
-          val intermediateAppJar = createIntermediateAppJar(
-              appVariant = appVariant,
-              diagnosticOutputDir = diagnosticOutputDir,
-              emitDebugInfo = extension.emitDebugInformation
-          )
-          val intermediateAndroidTestJar = createIntermediateAndroidTestJar(
-              diagnosticOutputDir = diagnosticOutputDir,
-              emitDebugInfo = extension.emitDebugInformation,
-              testVariant = this,
-              appJarsProvider = intermediateAppJar.flatMap { it.appJarsFile }
-          )
-          val inferAndroidTestUsageProvider = tasks.register(
-              "infer${name.capitalize(Locale.US)}KeepRulesForKeeper",
-              InferAndroidTestKeepRules(
-                  variantName = name,
-                  androidTestJarProvider = intermediateAndroidTestJar,
-                  releaseClassesJarProvider = intermediateAppJar,
-                  androidJar = androidJarRegularFileProvider,
-                  automaticallyAddR8Repo = extension.automaticR8RepoManagement,
-                  enableAssertions = extension.enableAssertions,
-                  extensionJvmArgs = extension.r8JvmArgs,
-                  r8Configuration = r8Configuration
-              )
-          )
+  private fun Project.configureKeepRulesGeneration(
+      appExtension: AppExtension,
+      extension: KeeperExtension,
+      diagnosticOutputDir: Provider<Directory>
+  ) {
+    // Set up r8 configuration
+    val r8Configuration = configurations.create(CONFIGURATION_NAME) {
+      description = "R8 dependencies for Keeper. This is used solely for the PrintUses CLI"
+      isVisible = false
+      isCanBeConsumed = false
+      isCanBeResolved = true
+      defaultDependencies {
+        add(project.dependencies.create("com.android.tools:r8:$DEFAULT_R8_VERSION"))
+      }
+    }
 
-          val prop = project.layout.dir(
-              inferAndroidTestUsageProvider.flatMap { it.outputProguardRules.asFile })
-          project.applyGeneratedRules(appVariant.name, prop)
+    val androidJarFileProvider = project.provider {
+      val compileSdkVersion = appExtension.compileSdkVersion
+          ?: error("No compileSdkVersion found")
+      File("${appExtension.sdkDirectory}/platforms/${compileSdkVersion}/android.jar").also {
+        check(it.exists()) {
+          "No android.jar found! Expected to find it at: $it"
         }
       }
     }
+    val androidJarRegularFileProvider = project.layout.file(androidJarFileProvider)
+
+    appExtension.testVariants.configureEach {
+      val appVariant = testedVariant
+      val extensionFilter = extension._variantFilter
+      val ignoredVariant = extensionFilter?.let {
+        project.logger.debug(
+          "$TAG Resolving ignored status for android variant ${appVariant.name}")
+        val filter = VariantFilterImpl(appVariant)
+        it.execute(filter)
+        project.logger.debug("$TAG Variant '${appVariant.name}' ignored? ${filter._ignored}")
```

Comment:
this is an extension function on `Project` already?

Reply:
54ead79